### PR TITLE
update validation and conversion status during upload

### PR DESCRIPTION
**Pull request by Bento007**
#### Reviewers
**Functional:** @mckinsel 

**Readability:** @MDunitz 

---

## Changes
- Add `DatasetAsset.create` to add assets to a dataset.
- Fix code that converts an ORM model to a dictionary with None values. If `remove_none` is True, continue, else If `remove_none` is False, add it to the returned dictionary.
- Add `generate_uuid` function to generate uuids to reduce code duplication.
- dataset_processing/process.py
  - Add updates to the database when conversion starts and finishes for cxg, seurat, loom, and h5ad.
  - Add updates to the database when validation starts and finishes for h5ad. 
  - Add `logger` to dataset_processing/process to log exceptions.
  - Modify `update_db` to use `processing_status_updater`
  - Moved s3_client to global scope
  - Add `create_artifact` functions to upload an artifact to s3 and update the dataset.
  - Localized variables to reduce the use of environment variables, and simplify testing.
  - Modify the return value of `convert_file_ignore_exceptions` to `file_name` and `ConversionStatus`
  - Add `get_bucket_prefix` function to reduce code duplication
  - Add `process_cxg` function to handle processing the CXG for code readability.
- Moved *generate_data_mixing* so it can be used in *test_process*
- Fix`GenerateDataMixin* to work even if there is no active session.

### CODE REVIEW

File: backend/corpora/common/utils/uuid.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import uuid
+
+
+def generate_uuid():
```
Comment (MDunitz) [inline backend/corpora/common/utils/uuid.py:4]: nit -- does this need to be in its own file? can it go in db_utils? or as a function on the base of the orm?

**Review by MDunitz — APPROVED** [review]
lgtm

File: backend/corpora/common/utils/uuid.py (hunk 1 above)
Comment (Bento007) [reply backend/corpora/common/utils/uuid.py:4]: I could put it in ORM, but it doesn't need to be apart of base.

**Review by mckinsel — APPROVED** [review]
LGTM. One tiny nit, and we otherwise discussed in slack.

**Pull request merged**